optimum/neuron/utils/cache_utils.py

```diff
@@ -564,13 +564,31 @@ class NeuronHash:
     tensor_parallel_size: Union[int, _UnspecifiedHashAttribute] = field(
         default_factory=_UnspecifiedHashAttribute.with_args(min_optimum_neuron_version="0.0.8", default=1)
     )
+    _is_private: Optional[bool] = None
+    _model_type: Optional[str] = None
     _hash: _MutableHashAttribute = field(default_factory=_MutableHashAttribute)
 
-    def __post_init__(self):
+    def __post_init__(self, model: "PreTrainedModel"):
         for attr in self.__dict__.values():
             if isinstance(attr, _UnspecifiedHashAttribute):
                 attr.check_requirements_are_met(self.neuron_compiler_version)
-        self.compute_hash()
+
+        # Checking whether the model is private or not.
+        is_private = None
+        model_name_or_path = _get_model_name_or_path(model.config)
+        if model_name_or_path is None:
+            is_private = True
+        elif Path(model_name_or_path).exists():
+            is_private = True
+        else:
+            is_private = is_private_repo(model_name_or_path)
+
+        # Using object.__setattr__ to change the field value because NeuronHash is supposed to be frozen.
+        # Not very clear, but it should work here.
+        super().__setattr__("_is_private", is_private)
+        super().__setattr__("_model_type", model.config.model_type)
```

Comment:
Not super clear indeed: all I can see is that these members are excluded from the hash computation down below ...
Can you clarify a bit ?

Reply:
You can refer to the notes that I have just added to the PR for more information (last point in the list), but basically storing a reference to the model in the `NeuronHash` was causing issues during compilation. 

It is actually not needed per-se to have a reference to the model so I got rid of it. Instead now `model` is annotated as an `InitVar`, meaning that it will be a parameter to the initializer of the `NeuronHash` dataclass, but not an attribute. 
During initialization, the model argument is then used to compute values that are needed by the class that were previously inferred from the model reference.  

The reason `super().__setattr__` is used here is because `NeuronHash` is a frozen dataclass (to make it hashable), but we need to set those values at post_init time... This is a bit of hack, but it should be ok here since we never change anything after initialization.